crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -1645,7 +1699,21 @@ impl<'src> Parser<'src> {
                     ))
                 }
             }
-            _ => Expr::Set(self.parse_set_expression(key_or_element.expr, start)),
+            _ => {
+                // test_err unparenthesized_walrus_set_literal_py39
+                // # parse_options: {"target-version": "3.9"}
+                // {x := 1, 2, 3}
+                // {1, x := 2, 3}
+                // {1, 2, x := 3}
+
+                if key_or_element.is_unparenthesized_named_expr() {
+                    self.add_unsupported_syntax_error(
+                        UnsupportedSyntaxErrorKind::UnparWalrus,
+                        key_or_element.range(),
+                    );
+                }
```

Comment:
Nit: I'm somewhat inclined to move this check into `parse_set_expression`. That would also move the tests closer to the actual logic (for the tail elements)

Reply:
I think I'd recommend moving the check to `parse_set_expression` where we have access to both the first and the rest of the set elements.